### DISCUSSION TITLE
Limit amount of ranked maps with the same song

### DISCUSSION THREAD

**Discussion by Tenexxt** [ppy/osu#29141] [Ideas] [answered]
Limit amount of ranked maps with the same song to be 3 or 4 for osu!std, and for other modes to 3 also.
The reason for that is because im sick of seeing the same songs getting ranked over and over.
My best example is song "Shikairo days" that has 11 ranked sets, and a lot still pending and possibly ranked in the future. Just for clousure - The map has 7 sets for !std, 1 for taiko, which is also a hybrid with !std, 1 for catch, and 3 for mania. I would understand if those 11 sets were equaly distributed between gamemodes, but come on, 7 sets for !std only?
I did some quick math by counting the guest diffs, and other people needed to get all those maps ranked.
Since 5 minimum hypes are required, thats 5. 2 BNs for nominations, thats 7 + beatmap host, so 8.
8 x 11 = 88
After i counted the guest difficulties for these maps, **assuming people don't repeat for basically everything in this equation** - we get a 42.
88 + 42 = 130.
**And thats a minimum**. Since maps 99% of the time need mods, and several bns focus on certain maps while they're qualified, to make sure everything is ok, that number could **VERY** easily go over 150.
Also, out of those 11 sets, only 1 is full size, rest is just the opening ver. (TV size)
While this song could be an exeption just as easily as not, the fact that this can happen is very stupid in my opinion.
We don't need a minimum of 130 people on the same song, come on.

**Comment by Stoppedpuma** [comment, marked as answer]
I think this is ridiculous as mappers should be able to map what they want and still have the ability of it being ranked if it meets requirements. Requiring a mapper to map to another ruleset (assuming it's still possible under your idea) is completely wrong in my eyes and forces mappers who want to have the map ranked to map under a ruleset they might not have experience in playing or mapping for. Additionally mapping styles change over time. Some may want a map to have a different style besides what's currently ranked. While an answer to this could be "just play graveyard" I think this is a completely invalid response as it takes away the ability to easily find maps of higher quality.

> The reason for that is because im sick of seeing the same songs getting ranked over and over.

They are popular songs, no one is forcing you to play them. I suggest you follow the idea behind your own words of "only 3-4 should be ranked" and try to find graveyard maps of songs you enjoy.

> I would understand if those 11 sets were equaly distributed between gamemodes, but come on, 7 sets for !std only?

osu! is majorily played by people who want to play the osu! ruleset. Other rulesets are simply not as popular as there are other games which people may prefer or the ruleset may be disliked by the player.

**Reply by Tenexxt** [reply]
While I don't like limiting people who want to map a song, 11 sets and possibly more in the future is ridiculous. While mapping requirements and styles change overtime, the sets were ranked within a span of a Month, and 4 sets are enough to give variety in my opinion.
Also, my idea doesn't block different versions of the song and remixes (unless they also reach the limit) all remixed songs and versions would have a seperate limit, so there could be 4 maps of a certain remix and 4 of a another one (example for !STD)
As for making unpopular songs, I'm currently working on 3 maps for mappers guild without a ranked set, and I'm planning to make more maps that don't have a ranked set.
Don't get me wrong, I don't like limiting people, but there are literally 10 TV size ranked sets with this song and only 1 is the full version. There is basically no variety. Tho I saw 1 remix while scrolling the pending category, which I still have to check out, but it's definitely better to make remixes than 10 same versions of the song

**Comment by cihe13375** [comment]
There was once a rule like "no two maps of the same song could exist in the qualified section at the same time", which is somewhat similar to the suggestion here but is not so harsh (it does not prevent 10 maps of the same song get ranked, but at least then cannot be ranked in the same week). However, in reality it lead to so called "speedrank wars" (mappers do whatever they can to get their own map qualified as fast as possible, and sometimes try to disqualify others' maps in order to get their own ones qualified). Finally mappers felt that it was just unnecessary so that rule was removed. If we hard limit the number of ranked maps of a song, the "speedrank" problem would probably become more severe, as now one's map will be graveyarded forever if they lose the war. So I don't see this happening.

However, I do agree that lots of maps of the same song lead to some issues:
1) reduce variety in the ranked section, because (1) the number of ranked maps per day is capped (<=10 per gamemode, and at least sometimes we do hit that limit), and (2) there is a minumum requirement of activity for BNs (so if they are not allowed to nominate song A and they need that nomination count to keep the BN status, then they have to nominate something else).
2) it can be confusing for users (including myself) who want to play that song: which mapset should I download? Most of rhythm games only have one set per song so for new comers it's natural to think that there is only one "genuine" set for the song. When I was a new player I usually only downloaded the one with the most play count, which may come from like 2010 (instead of a newer set which has fewer playcount but may have higher quality).

For 1), tbh idk if that's a huge problem, and if it was, how to solve it without introducing speedrank wars again. The <=10 cap is introduced for players who want to play every single ranked map (and also ensures every map does get some attention), so I don't see it being relaxed. Another option is to add a rule like "nominating repeating songs does not add to BN activity statistics" (similar to how a mega set with like 50 diffs is "same" as a regular 4 diff set), but that may lead to something like "speed nomination wars" which is worse than speedrank I guess...

For 2), a possible solution is grouping all maps of the same song as some kind of "superset", and the user can optionally treat them as a single mapset (e.g. display in a single webpage, download the whole superset at once, displayed as a single set in song select, etc.). I vaguely recall that similar ideas have been discussed before, but since it requires substantial coding/design/... work idk if it will be a thing. I think an easier approach is to add a "download all sets of this song / in search results" button in the map search page. It could also be useful for other purposes (e.g. "download all maps related to my favorite anime" or "download all maps of artist A's song")

**Reply by Tenexxt** [reply]
Personally - i don't think hard-limiting amount of maps is good, but basically there should be something that can prevent 10 and more maps with the same song and I didn't have any other idea.
The amount of songs from featured artists that don't have a map is sad. It took me barely any time to find multiple, so many were without a map that I actually had to decide which song to map, and I expected there to be less songs that I could decide from.
I recently started mapping for mappers guild and I'm planning to make maps for songs without them and push them for ranked because so many good music is not used and then we have 11 sets of the same song

**Reply by felipemarins** [reply]
> it can be confusing for users (including myself) who want to play that song: which mapset should I download?

I think it should be easily resolved with the upcoming beatmap tags as mentioned in the osu! youtube video from 1 month ago.
